src/v/features/feature_table.cc

```diff
@@ -697,6 +706,49 @@ void feature_table::assert_compatible_version(bool override) {
     }
 }
 
+feature_table::probe::probe(const feature_table& parent)
+  : _parent(parent) {}
+
+void feature_table::probe::setup_metrics() {
+    if (ss::this_shard_id() != 0) {
+        return;
+    }
+
+    if (!config::shard_local_cfg().disable_metrics()) {
+        setup_metrics_for(_metrics);
+    }
+
+    if (!config::shard_local_cfg().disable_public_metrics()) {
+        setup_metrics_for(_public_metrics);
+    }
+}
+
+long long feature_table::probe::calculate_expiry_metric(
+  const std::optional<security::license>& license,
+  security::license::clock::time_point now) {
+    if (!license) {
+        return -1;
+    }
```

Comment:
discussion:

If there is no license, it may be possible to not set up this metric (or even the probe), but then setting up the probe or metric must be attached to `set_license` / `revoke_license / `feature_table_snapshot::apply`.

For the latter, it may be worth:
```diff
diff --git a/src/v/features/feature_table_snapshot.cc b/src/v/features/feature_table_snapshot.cc
index 6e1cfb406e..b5477a88c5 100644
--- a/src/v/features/feature_table_snapshot.cc
+++ b/src/v/features/feature_table_snapshot.cc
@@ -42,7 +42,11 @@ void feature_table_snapshot::apply(feature_table& ft) const {
       applied_offset);
 
     ft.set_active_version(version);
-    ft._license = license;
+    if (license) {
+        ft.set_license(*license);
+    } else {
+        ft.revoke_license();
+    }
 ```

Reply:
Yeah, that's a good point to discuss but I've kept this as is for now to keep things simpler. I kinda like having the -1 explicitly present to signal that the license it unset (as opposed to just the metric missing because of network issues, Prometheus scraper being down, etc.). But I don't feel strongly either way which makes me prefer simplicity.